tests/pipelinesrun_test.go

```diff
@@ -0,0 +1,209 @@
+	if *PipelinesVcsToken == "" {
+		assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured")
+		return
+	}
+	// Create integration with provided token.'
+	unixTime := time.Now().Unix()
+	timeString := strconv.Itoa(int(unixTime))
+	integrationName := strings.Join([]string{"github", "integration_test", timeString}, "_")
+	integrationId, err := testsPipelinesIntegrationsService.CreateGithubIntegration(integrationName, *PipelinesVcsToken)
+	assert.NoError(t, err)
+	defer deleteIntegrationAndAssert(t, integrationId)
+
+	// Create source with the above integration and assert.
+	sourceId, srcErr := testsPipelinesSourcesService.AddSource(integrationId, *PipelinesVcsRepoFullPath, *PipelinesVcsBranch, services.DefaultPipelinesFileFilter)
+	assert.NoError(t, srcErr)
+
+	defer deleteSourceAndAssert(t, sourceId)
+	pollSyncPipelineSource(t)
+	pollForSyncResourceStatus(t)
+}
+
+func testGetRunStatus(t *testing.T) {
+	assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured")
+	// Create integration with provided token.
+	unixTime := time.Now().Unix()
+	timeString := strconv.Itoa(int(unixTime))
+	integrationName := strings.Join([]string{"github", "integration_test", timeString}, "_")
+	integrationId, err := testsPipelinesIntegrationsService.CreateGithubIntegration(integrationName, *PipelinesVcsToken)
+	assert.NoError(t, err)
+
+	defer deleteIntegrationAndAssert(t, integrationId)
+
+	// Create source with the above integration and assert.
+	sourceId, sourceErr := testsPipelinesSourcesService.AddSource(integrationId, *PipelinesVcsRepoFullPath, *PipelinesVcsBranch, services.DefaultPipelinesFileFilter)
+	assert.NoError(t, sourceErr)
+	defer deleteSourceAndAssert(t, sourceId)
+
+	pollSyncPipelineSource(t)
+
+	pollForSyncResourceStatus(t)
+	_, isMultiBranch, resourceErr := testPipelinesSyncService.GetPipelineResourceID(*PipelinesVcsRepoFullPath)
+	assert.NoError(t, resourceErr)
+	pipelineName := "pipelines_run_int_test"
+	status, trigErr := testPipelinesRunService.TriggerPipelineRun(*PipelinesVcsBranch, pipelineName, isMultiBranch)
+	assert.NoError(t, trigErr)
+	assertTriggerRun(t, pipelineName, *PipelinesVcsBranch, status)
+
+	pollGetRunStatus(t, pipelineName)
+}
+
+func pollGetRunStatus(t *testing.T, pipelineName string) {
+	pollingAction := func() (shouldStop bool, responseBody []byte, err error) {
+		_, isMultiBranch, resourceErr := testPipelinesSyncService.GetPipelineResourceID(*PipelinesVcsRepoFullPath)
+		assert.NoError(t, resourceErr)
+		pipRunResponse, syncErr := testPipelinesRunService.GetRunStatus(*PipelinesVcsBranch, pipelineName, isMultiBranch)
+		assert.NoError(t, syncErr)
+
+		// Got the full valid response.
+		if pipRunResponse != nil && len(pipRunResponse.Pipelines) > 0 && pipRunResponse.Pipelines[0].Name == pipelineName {
+			log.Info("pipelines status code ", pipRunResponse.Pipelines[0].Run.StatusCode)
+			if isCancellable(pipRunResponse.Pipelines[0].Run.StatusCode) {
+
```

Comment:
```suggestion
```

Reply:
didn't get this I need to check whether a pipeline is in cancellable state before sending cancel request

Reply:
Just remove the empty line :)

Reply:
I thought the whole cancellation check had to be removed 👍 I will change